R/diff_secuTrial.R

```diff
@@ -0,0 +1,58 @@
+#' Returns differences in the setup of two secuTrialdata objects
+#' @description During ongoing studies it is possible that changes to the DataCapture interface
+#'              are made. Sometimes these changes may call for adjustments in analysis code.
+#'              It is considered good practice to run \code{diff_secuTrial()} on the last export
+#'              and the current export of a project to at least make yourself aware of
+#'              potential changes to the setup. If there are differences, the results of this function should
+#'              be interpreted as a first indicator since they may not cover all alterations.
+#'              Information is returned on new forms and variables.
+#'              A detailed list of changes can be produced in the FormBuilder with
+#'              "Compare project setup".
+#' @param x a \code{secuTrialdata} object (the older export)
+#' @param y a \code{secuTrialdata} object (the newer export)
+#' @export
+#' @details If there are differences, \code{diff_secuTrial()} will produce a list of vectors.
+#'          The fist vector informs about new forms and the second vector informs about
+#'          new variables.
+#'
+#' @examples
+#' # export location
+#' export_loc <- system.file("extdata", "sT_exports", "longnames",
+#'                           "s_export_CSV-xls_CTU05_long_ref_miss_en_utf8.zip",
+#'                           package = "secuTrialR")
+#' # read export
+#' sT_export <- read_secuTrial(export_loc)
+#'
+#' # return scores // in this case on the same export
+#' diff_secuTrial(sT_export, sT_export)
+#'
+diff_secuTrial <- function(x, y) {
+  if (class(x) == "secuTrialdata" & class(y) == "secuTrialdata") {
+
+    # comapring different projects makes no sense really
+    if(x$export_options$project_name != y$export_options$project_name) {
+      stop("The two exports appear to originate from different secuTrial projects.")
+    }
+
+    if(x$export_options$project_version == y$export_options$project_version) {
+      print(paste0("The project structure has not changed. Project versions are both: ",
+                   x$export_options$project_version))
+    } else if (x$export_options$project_version != y$export_options$project_version) {
+      # forms
+      forms_x <- x[[x$export_options$meta_names$forms]]
+      forms_y <- y[[y$export_options$meta_names$forms]]
+      form_diff <- unique(forms_y$formtablename[which(! forms_y$formtablename %in% forms_x$formtablename)])
+
+      # items
+      items_x <- x[[x$export_options$meta_names$items]]
+      items_y <- y[[y$export_options$meta_names$items]]
+      item_diff <- unique(items_y$ffcolname[which(! items_y$ffcolname %in% items_x$ffcolname)])
```

Comment:
Can one remove columns from secuTrial? If so, it might be usefull to know which columns were removed.

Reply:
I will clarify this and if yes will add code to report on removed variables.

Reply:
You can not remove them. The only option is to hide then and then choose not to export hidden data. I write a brief "Did you know..." in the secuTrial recipes (currently the top one).

https://swissclinicaltrialorganisation.github.io/secuTrial_recipes/did_you_know/

Thus, I think one way comparison should be fine?

Reply:
Would a scenario like this one be possible:
1. export data from a database
2. change project version by hiding a field
3. export data without hidden fields from the database
Then the two exports would stem from different versions, but the diff wouldn't show that the hidden fields are missing from the second export. If that is a possibility, then a two-way check would be better.

Reply:
Correct.